### PR TITLE
Pipeline int tests fix

### PR DESCRIPTION
**Pull request by kat1906**

### CODE REVIEW

File: app/test-utils/testFunctions/questionTests/errorTests.js [hunk 1]
```diff
@@ -86,7 +86,8 @@ const errorTests = ({
       .expect(errorSummaryList.find('li:nth-child(1)').innerText).eql(expectedErrorMessage)
       .expect(errorSummaryList.find('li:nth-child(1) a').getAttribute('href')).eql(`${expectedAnchorLink}`)
       .expect(renderedQuestion.exists).ok()
-      .expect(renderedQuestion.find('.nhsuk-error-message').innerText).eql(`Error:\n${expectedErrorMessage}`);
+      .expect(renderedQuestion.find('.nhsuk-error-message').innerText).contains('Error:')
+      .expect(renderedQuestion.find('.nhsuk-error-message').innerText).contains(expectedErrorMessage);
```
Comment (kat1906) [inline app/test-utils/testFunctions/questionTests/errorTests.js]: The pipeline tests are showing as a space and locally as \n so I have split this test into 2 parts. Also had to use contains instead pf trimming as it is found after the button is clicked so cannot be found at the start of the tests.

File: app/dashboard/subDashboards/browser-based/ui.test.js [hunk 2]
```diff
@@ -16,13 +16,6 @@ const pageSetup = async (t) => {
 };
 
 fixture('Show browser based dashboard page');
-// .afterEach(async (t) => {
-// const isDone = nock.isDone();
-// if (!isDone) {
-//   nock.cleanAll();
-// }
-// await t.expect(isDone).ok('Not all nock interceptors were used!');
-// });
```
Comment (kat1906) [inline app/dashboard/subDashboards/browser-based/ui.test.js]: removed these for now

File: app/preview/ui.test.js [hunk 3]
```diff
@@ -123,13 +117,12 @@ test('when existing marketing data - The client application type section and bro
...
   const hardwareRequirementsDescriptionRow = browserBasedExpandaleSectionTable.find('[data-test-id="view-section-table-row-hardware-requirements-description"]');
   const additionalInformationRow = browserBasedExpandaleSectionTable.find('[data-test-id="view-section-table-row-additional-information"]');
-
   await t
     .expect(clientApplicationTypesSection.exists).ok()
     .expect(clientApplicationTypesSection.find('h3').innerText).eql('Client application type')
 
     .expect(browserBasedExpandableSection.exists).ok()
-    .expect(browserBasedExpandableSection.innerText).eql('Browser based application')
+    .expect(browserBasedExpandableSectionText.trim()).eql('Browser based application')
```
Comment (kat1906) [inline app/preview/ui.test.js]: might need to do the rest of the innerText expects after this one as they might fail after this one passes

File: app/test-utils/testFunctions/questionTests/errorTests.js (hunk 1 above)
Comment (alibombaye) [reply app/test-utils/testFunctions/questionTests/errorTests.js]: 👍

**Comment by alibombaye** [comment]
looks great Katie. Once we know that this fix is working and doing the job. Should we perhaps to do this for everything that we do an innerText for? well perhaps just the label not the populated text. Just to keep consistent.

**Comment by kat1906** [comment]
I very much expect they will fail anyway so might as well do it now I guess!!

**Comment by kat1906** [comment]
(I mean the tests outside the section tests)

**Comment by alibombaye** [comment]
Do the ones you have done so far pass on the pipeline?
Could we create like a little helper function to make this a little less painful

**Comment by kat1906** [comment]
> Do the ones you have done so far pass on the pipeline?
> Could we create like a little helper function to make this a little less painful

Yes the merged ones work in pipeline.
The problem is really only in the preview tests now ... which are painful anyway :(

**Comment by alibombaye** [comment]
> > Do the ones you have done so far pass on the pipeline?
> > Could we create like a little helper function to make this a little less painful
> 
> Yes the merged ones work in pipeline.
> The problem is really only in the preview tests now ... which are painful anyway :(

Oooh I didn't know you could quote a reply, Good that innit

File: app/preview/ui.test.js [hunk 4]
```diff
@@ -176,76 +188,106 @@ test('when existing marketing data - The client application type section and nat
...
+  const deviceCapabilitiesRowTitle = await deviceCapabilitiesRow.find('div[data-test-id="view-section-table-row-title"]').innerText;
+  const deviceCapabilitiesRowItems = await deviceCapabilitiesRow.find('div[data-test-id="view-section-table-row-component"]').innerText;
   const hardwareRequirementsRow = nativeMobileExpandaleSectionTable.find('[data-test-id="view-section-table-row-hardware-requirements"]');
+  const hardwareRequirementsRowTitle = await hardwareRequirementsRow.find('div[data-test-id="view-section-table-row-title"]').innerText;
+  const hardwareRequirementsRowItems = await hardwareRequirementsRow.find('div[data-test-id="view-section-table-row-component"]').innerText;
   const additionalInformationRow = nativeMobileExpandaleSectionTable.find('[data-test-id="view-section-table-row-additional-information"]');
+  const additionalInformationRowTitle = await additionalInformationRow.find('div[data-test-id="view-section-table-row-title"]').innerText;
+  const additionalInformationRowItems = await additionalInformationRow.find('div[data-test-id="view-section-table-row-component"]').innerText;
```
Comment (alibombaye) [inline app/preview/ui.test.js]: YIKES!!! 😱 
Lets stick with this for now. But looks kinda painful

**Pull request merged**